src/preload/syscall_hook.S

```diff
@@ -572,6 +572,440 @@ SYSCALLHOOK_START(_syscall_hook_trampoline_49_89_ca)
+        ldp     q20, q21, [sp, 496]
+        .cfi_restore q20
+        .cfi_restore q21
+        ldp     q22, q23, [sp, 528]
+        .cfi_restore q22
+        .cfi_restore q23
+        ldp     q24, q25, [sp, 560]
+        .cfi_restore q24
+        .cfi_restore q25
+        ldp     q26, q27, [sp, 592]
+        .cfi_restore q26
+        .cfi_restore q27
+        ldp     q28, q29, [sp, 624]
+        .cfi_restore q28
+        .cfi_restore q29
+        ldp     q30, q31, [sp, 656]
+        .cfi_restore q30
+        .cfi_restore q31
+
+        ldp     q0, q1, [sp], 688
+        .cfi_restore q0
+        .cfi_restore q1
+        .cfi_def_cfa_offset 0
+        ret
+        .cfi_endproc
+        .size   _syscall_hook_trampoline, .-_syscall_hook_trampoline
+
+/**
+ * On syscallhook entry, we are still on the old stack,
+ * with x30 (lr) points to right after the blr instruction that got us here.
+ * The old values of x15 and x30 are saved to [x8], which is the syscall number
+ * with an offset to land in the stub_scratch_2 area.
+ */
+        .globl _syscall_hook_trampoline_raw
+        .hidden _syscall_hook_trampoline_raw
+        .type _syscall_hook_trampoline_raw, @function
+_syscall_hook_trampoline_raw:
+        .cfi_startproc
+        .cfi_return_column 32 // pc
+        .cfi_escape     0x10, /* DW_CFA_expression */   \
+                        0x20, /* pc */                  \
+                        0x02, /* length 2 */            \
+                        0x8e, /* DW_OP_breg30 */        \
+                        0x00
+        .cfi_escape     0x10, /* DW_CFA_expression */   \
+                        0x0f, /* x15 */                 \
+                        0x03, /* length 3 */            \
+                        0x78, /* DW_OP_breg8 */         \
+                        (stub_scratch_2 - preload_thread_locals) | 0x80, \
+                        0x00
+        .cfi_escape     0x10, /* DW_CFA_expression */   \
+                        0x1e, /* x30 */                 \
+                        0x03, /* length 3 */            \
+                        0x78, /* DW_OP_breg8 */         \
+                        (stub_scratch_2 - preload_thread_locals + 8) | 0x80, \
+                        0x00
+        .cfi_escape     0x16, /* DW_CFA_val_expression */                       \
+                        0x08, /* x8 */                                          \
+                        0x06, /* length 6 */                                    \
+                        0x78, /* DW_OP_breg8 */                                 \
+                        ((-preload_thread_locals) & 0x7F) | 0x80,               \
+                        ((-preload_thread_locals) & (0x7F << 7)) >> 7 | 0x80,   \
+                        ((-preload_thread_locals) & (0x7F << 14)) >> 14 | 0x80, \
+                        ((-preload_thread_locals) & (0x7F << 21)) >> 21 | 0x80, \
+                        ((-preload_thread_locals) & (0x7F << 28)) >> 28 | 0x70
```

Comment:
Why is the `0x70` here required?

Reply:
I don't remember how generic this encoding was but I'm pretty sure this is to get the sign of the LEB128 right...

Reply:
Yeah I think these are the sign bits I need to get a 35bits negative number from a 32bit negative number. I might be able to define a constant like `-preload_thread_locals | (7 << 32)` and use that in all the expressions. I wasn't sure if I need to worry about integer type though (that constant must not be 32bit) which is why I didn't do it that way...

Reply:
Ok. That's reasonable, but the please
A) adjust the mask for only the bits you pick up to make it clear and
B) add a comment